src/univariate/discrete/soliton.jl

```diff
@@ -0,0 +1,119 @@
+"""
+
+Soliton(K::Integer, M::Integer, δ::Real, atol::Real=0) <: Distribution{Univariate, Discrete}
+
+The Robust Soliton distribution of length `K`, mode `M` (i.e., the
+location of the robust component spike), peeling process failure
+probability `δ`, and minimum non-zero probability mass `atol`. More
+specifically, degrees `i` for which `pdf(Ω, i)<atol` are set to
+`0`. Letting `atol=0` yields the regular robust Soliton distribution.
+
+```julia
+Soliton(K, M, δ)        # Robust Soliton distribution (with atol=0)
+Soliton(K, M, δ, atol)  # Robust Soliton distribution with minimum non-zero probability mass atol
+
+params(Ω)               # Get the parameters ,i.e., (K, M, δ, atol)
+degrees(Ω)              # Return a vector composed of the degrees with non-zero probability mass
+pdf(Ω, i)               # Evaluate the pdf at i
+cdf(Ω, i)               # Evaluate the pdf at i
+rand(Ω)                 # Sample from Ω
+rand(Ω, n)              # Draw n samples from Ω
+```
+
+External links:
+
+* [Soliton distribution on Wikipedia](https://en.wikipedia.org/wiki/Soliton_distribution)
+
+"""
+struct Soliton <: DiscreteUnivariateDistribution
+    K::Int64 # Number of input symbols
+    M::Int64 # Location of the robust component spike
+    δ::Float64 # Peeling process failure probability
+    atol::Float64 # Minimum non-zero probability assigned to a degree
+    degrees::Vector{Int} # Degrees with non-zero probability
+    CDF::Vector{Float64} # CDF evaluated at each element in degrees
+    function Soliton(K::Integer, M::Integer, δ::Real, atol::Real=0)
+        0 < K || throw(ArgumentError("Expected 0 < K, but got $K."))
+        0 < δ < 1 || throw(ArgumentError("Expected 0 < δ < 1, but got $δ."))
+        0 < M <= K || throw(ArgumentError("Expected 0 < M <= K, but got $M."))
+        0 <= atol < 1 || throw(ArgumentError("Expected 0 <= atol < 1, but got $atol."))
```

Comment:
Can/should all these by `DomainError`s?

```suggestion
        0 <= atol < 1 || throw(DomainError(atol, "Expected 0 <= atol < 1."))
```

Reply:
I agree. I've updated the code.